test/e2e/defaulting.go

```diff
@@ -0,0 +1,174 @@
+			}
+			qualifiedName = &util.QualifiedName{
+				Name: kubeFedConfigName,
+			}
+		}
+
+		if defaultKubeFedConfig == nil {
+			defaultKubeFedConfig = &v1beta1.KubeFedConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: qualifiedName.Name,
+				},
+				Spec: v1beta1.KubeFedConfigSpec{
+					// Set some values different from the defaults used. This
+					// ensures they are retained after defaulting is applied.
+					ControllerDuration: &v1beta1.DurationConfig{
+						AvailableDelay: &metav1.Duration{
+							Duration: defaults.DefaultClusterAvailableDelay + 11*time.Second,
+						},
+					},
+					FeatureGates: []v1beta1.FeatureGatesConfig{
+						{
+							Name:          string(features.PushReconciler),
+							Configuration: v1beta1.ConfigurationDisabled,
+						},
+					},
+					ClusterHealthCheck: &v1beta1.ClusterHealthCheckConfig{
+
+						Period: &metav1.Duration{
+							Duration: defaults.DefaultClusterHealthCheckPeriod + 11*time.Second,
+						},
+					},
+					Scope: apiextv1b1.ClusterScoped, // Required
+				},
+			}
+		}
+
+		// Ensure these get set for every test even if already initialized.
+		qualifiedName.Namespace = f.TestNamespaceName()
+		defaultKubeFedConfig.ObjectMeta.Namespace = qualifiedName.Namespace
+	})
+
+	It(fmt.Sprintf("%s is created by mutating admissing webhook and preserves values set by user", resourceName), func() {
+		By(fmt.Sprintf("Creating a %s defaulted by the mutating admission webhook", resourceName))
+		webhookDefaultedKubeFedConfig := defaultKubeFedConfig.DeepCopyObject().(*v1beta1.KubeFedConfig)
+		err := client.Create(context.TODO(), webhookDefaultedKubeFedConfig)
+		framework.ExpectNoError(err, fmt.Sprintf("Error creating default %q", *qualifiedName))
+
+		By(fmt.Sprintf("Creating a %s defaulted explicitly in this test", resourceName))
+		myDefaultedKubeFedConfig := defaultKubeFedConfig.DeepCopyObject().(*v1beta1.KubeFedConfig)
+		defaults.SetDefaultKubeFedConfig(myDefaultedKubeFedConfig)
+
+		By(fmt.Sprintf("Verifying the %s defaulted by the mutating admission webhook matches the one defaulted explicitly in this test", resourceName))
+		common.WaitForObject(tl, qualifiedName.Namespace, qualifiedName.Name, kubeFedConfigGetter, myDefaultedKubeFedConfig, framework.PollInterval, framework.TestContext.SingleCallTimeout, util.ObjectMetaAndSpecEquivalent)
+
+		if framework.TestContext.LimitedScope {
+			// Delete the KubeFedConfig we created since the kubefed system
+			// namespace will not be deleted when running namespace scoped.
+			framework.AddCleanupAction(func() {
+				fedConfig := webhookDefaultedKubeFedConfig
+				err := client.Delete(context.TODO(), fedConfig, fedConfig.Namespace, fedConfig.Name)
+				if err != nil && !apierrors.IsNotFound(err) {
+					f.Logger().Errorf("Error deleting %s %q: %v", resourceName, *qualifiedName, err)
+				}
+			})
+		}
+	})
+
+	// TODO(font): This test explicitly verifies the controller-manager detects
```

Comment:
If the crds and admission webhook were deployed via a subchart whose installation was guaranteed to be complete before deployment of the controller manager, would #983 still be a problem?